### PR TITLE
[MRG, ENH] Find aseg labels near montage

### PR DESCRIPTION
**Pull request by alexrockhill**
The idea here is to find which voxels are nearby an electrode contact in order to know which surfaces to plot and for analysis.

This shares internal functions with https://github.com/mne-tools/mne-python/pull/9544/ and is a good reason to make `mne.surfaces._find_voxel_neighbors` private so that it can be used in both scenarios without confusion and user-facing function.

### CODE REVIEW

**Comment by alexrockhill** [comment]
@larsoner, I think it might be best to change the default back to `aseg.mgz` because `fsaverage` doesn't have any other segmentations in the sample data.

**Comment by alexrockhill** [comment]
This is ready for review too, but this PR has to merge first https://github.com/mne-tools/mne-misc-data/pull/9

**Comment by larsoner** [comment]
Even after fetch_fsaverage is called?

We can also update the sample dataset pretty easily, so I wouldn't let it be a limiting factor

**Comment by alexrockhill** [comment]
> Even after fetch_fsaverage is called?
> 
> We can also update the sample dataset pretty easily, so I wouldn't let it be a limiting factor

Yes but good to know, maybe we should update that then

**Comment by alexrockhill** [comment]
This also updates the seeg tutorial which was out-of-date because of the new data format

**Comment by alexrockhill** [comment]
Here's what the plots I added look like by the way.

Screen Shot 2021-07-14 at 8 23 04 PM

**Comment by larsoner** [comment]
>  I think it might be best to change the default back to aseg.mgz because fsaverage doesn't have any other segmentations in the sample data.

I just did `mne.datasets.sample.data_path(force_update=True, verbose=True)` (i.e., not even running `fetch_fsaverage`) and I see:
```
$ ls ~/mne_data/MNE-sample-data/subjects/fsaverage/mri
aparc.a2005s+aseg.mgz  aparc+aseg.mgz  brainmask.mgz  lh.ribbon.mgz   orig      p.aseg.mgz     ribbon.mgz        subcort.prob.mgz  transforms
aparc.a2009s+aseg.mgz  aseg.mgz        brain.mgz      mni305.cor.mgz  orig.mgz  rh.ribbon.mgz  subcort.prob.log  T1.mgz
```
so `aparc+aseg.mgz` is in there. Maybe your version of `sample` was outdated?

**Review by larsoner — COMMENTED** [review]
Can anything be refactored with `mne.label.find_pos_in_annot`?

**Comment by alexrockhill** [comment]
> find_pos_in_annot

I didn't see that one, yeah looks like it. I think it might have to be refactored to accept multiple positions though since you wouldn't want to be loading the aseg file over and over.

**Comment by alexrockhill** [comment]
> > I think it might be best to change the default back to aseg.mgz because fsaverage doesn't have any other segmentations in the sample data.
> 
> I just did `mne.datasets.sample.data_path(force_update=True, verbose=True)` (i.e., not even running `fetch_fsaverage`) and I see:
> 
> ```
> $ ls ~/mne_data/MNE-sample-data/subjects/fsaverage/mri
> aparc.a2005s+aseg.mgz  aparc+aseg.mgz  brainmask.mgz  lh.ribbon.mgz   orig      p.aseg.mgz     ribbon.mgz        subcort.prob.mgz  transforms
> aparc.a2009s+aseg.mgz  aseg.mgz        brain.mgz      mni305.cor.mgz  orig.mgz  rh.ribbon.mgz  subcort.prob.log  T1.mgz
> ```
> 
> so `aparc+aseg.mgz` is in there. Maybe your version of `sample` was outdated?

Yep, mine must have just been way out-of-date.

**Comment by alexrockhill** [comment]
Ok so I changed one thing that I was very unsure about: `label.get_pos_in_annot` went to `label.get_pos_in_aseg` which included backdating a function by someone else. Since it was before the next release, I think this might be okay since I think calling it `annot` is confusing because `annot` files are something slightly different and already in use to mean a different thing.

Other than that, I tried to refactor to use the function but realized after doing so that it actually would be really inefficient because each channel of the montage is looped over so the aseg would have to be loaded in every time to use that function. I did refactor a bit into a `_get_aseg` function in `_freesurfer.py` so that some of the codebase is shared.

If all that is okay, should be good to go.

**Comment by larsoner** [comment]
> label.get_pos_in_aseg which included backdating a function by someone else. Since it was before the next release, I think this might be okay since I think calling it annot is confusing because annot files are something slightly different and already in use to mean a different thing.

Agreed this naming is better, I'll look at the refactoring now

**Comment by larsoner** [comment]
Actually I'll wait until #9544 is merged and this is rebased, feel free to ping me once that's done @alexrockhill

**Comment by alexrockhill** [comment]
I think that looks pretty cool and informative

Screen Shot 2021-07-16 at 1 18 59 PM

**Comment by alexrockhill** [comment]
> > label.get_pos_in_aseg which included backdating a function by someone else. Since it was before the next release, I think this might be okay since I think calling it annot is confusing because annot files are something slightly different and already in use to mean a different thing.
> 
> Agreed this naming is better, I'll look at the refactoring now

I took that part out and put it in a separate PR, this should be good to go as well after I rebase

**Comment by larsoner** [comment]
> I took that part out and put it in a separate PR, this should be good to go as well after I rebase

The refactoring PR should be merged first, then, right?

**Comment by alexrockhill** [comment]
> > I took that part out and put it in a separate PR, this should be good to go as well after I rebase
> 
> The refactoring PR should be merged first, then, right?

They are actually independent/use the exact same `_freesurfer.py` `_get_aseg` code from a `git checkout aseg -- mne/_freesurfer.py` call.

**Comment by alexrockhill** [comment]
Looks good to merge by me!

File: doc/changes/latest.inc
Comment (agramfort) [inline doc/changes/latest.inc:112]: Why tend to calls ROIs labels everywhere in mne / freesurfer world. So here it would be volume labels?

Get_montage_volume_labels or get_montage_vol_labels ?

File: mne/surface.py
Comment (agramfort) [inline mne/surface.py:1894]: Say where the ROIs / volume labels come from. Say it’s coming from a freesurfer volume parcellation

File: mne/surface.py
Comment (agramfort) [inline mne/surface.py:1913]: Please check of the acronym ROI is used anywhere in mne and use caps for an acronym when not a variable name.

File: mne/viz/circle.py
Comment (agramfort) [inline mne/viz/circle.py:431]: Can you say here what is the rationale for this function. I would suggest to say it makes sense for sEEG channels a not any type of channels.

File: mne/viz/circle.py
Comment (agramfort) [inline mne/viz/circle.py:441]: I prefer to avoid kwargs if possible.

File: doc/changes/latest.inc
Comment (larsoner) [reply doc/changes/latest.inc:112]: + 1 for `get_montage_volume_labels`

File: mne/surface.py
Comment (alexrockhill) [reply mne/surface.py:1913]: It's used here https://github.com/mne-tools/mne-python/blob/main/examples/visualization/roi_erpimage_by_rt.py but it looks maybe an EEGLAB thing to mean general regions of sensor space (also several places here https://github.com/mne-tools/mne-python/blob/main/mne/channels/channels.py).

Also here in matlab.rst

```
 +--------------------------------+--------------------------------------------------------------+
 | mne_read_label_file            | Read a label file (ROI).                                     |
 +--------------------------------+--------------------------------------------------------------+
 | mne_write_label_file           | Write a label file (ROI).                                    |
 +--------------------------------+--------------------------------------------------------------+
```

And finally also the one that I assume you were thinking of is in several locations, which is using it synonymously with label.

```
# load the anatomical ROI for comparison
anat_label = mne.read_labels_from_annot(subject, parc='aparc',
                                       subjects_dir=subjects_dir,
```

I'll just go ahead and change to `label`.

File: mne/viz/circle.py
Comment (alexrockhill) [reply mne/viz/circle.py:431]: I think you could use it for any channel type provided you have some labels/classes associated with it but I agree there are not a lot of other use cases.

File: mne/viz/circle.py
Comment (alexrockhill) [reply mne/viz/circle.py:441]: There are 24 arguments that could possibly be used from `mne.viz.plot_connectivity_circle` to change the plotting style so I'm not sure this is possible. It especially provides the ability to change fontsizes and colors which is really nice for plots.

File: mne/viz/circle.py
Comment (agramfort) [reply mne/viz/circle.py:431]: Well the pb I see is that field propagation is not isotropic  ie distance to source is not monotic with field strength. I fear such a function without a warning can enourage the type of false conclusions you can see in certain EEG papers.

File: mne/viz/circle.py
Comment (agramfort) [reply mne/viz/circle.py:445]: Hum… let’s say I am getting old and more flexible… I can live with that

**Review by agramfort — APPROVED** [review]

**Comment by larsoner** [comment]
Thanks @alexrockhill !

**Comment by drammock** [comment]
@alexrockhill The doc build failed on this PR, and it looks related, see https://app.circleci.com/pipelines/github/mne-tools/mne-python/9131/workflows/e0f6c14c-27ff-43dd-90c2-939a03863e81/jobs/31255

**Comment by larsoner** [comment]
@alexrockhill can you add a fix to #9540 ? I think that's very close to merge

**Comment by alexrockhill** [comment]
> @alexrockhill can you add a fix to #9540 ? I think that's very close to merge

Sure right on it

**Pull request merged**
